### ISSUE TITLE
filter_alignment.py error with entropy filtering qiime 1.9.0-rc1

### ISSUE THREAD

**Issue by ekopylova** [biocore/qiime#1824]
@gregcaporaso, running filter_alignment.py with default parameters and --suppress_lane_mask_filter works well. Adding the entropy parameter -e causes the program to crash (example below). This error does not seem to appear in Qiime 1.8.0. 

```bash
filter_alignment.py -o pynast_aligned_seqs -i pynast_aligned_seqs/rep_set_aligned.fasta -e 0.005 --suppress_lane_mask_filter
warnings.warn(_use_error_msg)
Traceback (most recent call last):
  File "/Users/jenya/install_dir/qiime/scripts/filter_alignment.py", line 149, in <module>
    main()
  File "/Users/jenya/install_dir/qiime/scripts/filter_alignment.py", line 142, in main
    entropy_threshold=opts.entropy_threshold):
  File "/Users/jenya/install_dir/qiime/qiime/filter_alignment.py", line 122, in apply_lane_mask_and_gap_filter
    seq = entropy_filter_f(seq)
  File "/Users/jenya/install_dir/qiime/qiime/filter_alignment.py", line 106, in <lambda>
    entropy_filter_f = lambda x: get_masked_string(x, ent_mask)
  File "/Users/jenya/install_dir/qiime/qiime/filter_alignment.py", line 38, in get_masked_string
    return (fromstring(s, dtype=uint8))[p].tostring()
IndexError: index 1225 is out of bounds for size 1225
```

**Comment by gregcaporaso** [comment]
Thanks for testing the release candidate! I'll look into this tomorrow.

**Comment by gregcaporaso** [comment]
@ekopylova, we have a fix in place for this now (#1825). We confirmed that this works with the example file that you sent us (but note that the result will be slightly different than the result that you got from QIIME 1.8.0, because we modified how highly entropic positions are identified in this pull request).

**Comment by jairideout** [comment]
@ekopylova this has been merged. I tested your data on QIIME 1.7.0 AMI and we get identical results to the output generated by the changes in this pull request (the only differences are sequence descriptions added to the output FASTA file, which aren't important).
